libraries/core_libs/consensus/include/vote_manager/vote_manager.hpp

```diff
@@ -152,70 +136,35 @@ class VoteManager {
    */
   void setNetwork(std::weak_ptr<Network> network);
 
-  // Unverified votes
-  /**
-   * @brief Add a vote to the unverified votes map
-   * @param vote vote
-   * @return true if added
-   */
-  bool addUnverifiedVote(std::shared_ptr<Vote> const& vote);
-
-  /**
-   * @brief Add verified votes to the unverified votes map
-   * @param votes verified votes
-   */
-  void moveVerifyToUnverify(std::vector<std::shared_ptr<Vote>> const& votes);
-
-  /**
-   * @brief Remove an unverified vote
-   * @param pbft_round vote PBFT round
-   * @param vote_hash vote hash
-   */
-  void removeUnverifiedVote(uint64_t pbft_round, vote_hash_t const& vote_hash);
-
-  /**
-   * @brief Check if the vote has been in the unverified votes map
-   * @param pbft_round vote PBFT round
-   * @param vote_hash vote hash
-   * @return true if exist
-   */
-  bool voteInUnverifiedMap(uint64_t pbft_round, vote_hash_t const& vote_hash);
-
-  /**
-   * @brief Get all unverified votes
-   * @return all unverified votes
-   */
-  std::vector<std::shared_ptr<Vote>> copyUnverifiedVotes();
-
-  /**
-   * @brief Clear the unverified votes map
-   */
-  void clearUnverifiedVotesTable();
-
-  /**
-   * @brief Get the total size of all unverified votes
-   * @return the total size of all unverified votes
-   */
-  uint64_t getUnverifiedVotesSize() const;
-
   // Verified votes
   /**
    * @brief Add a vote to the verified votes map
    * @param vote vote
+   *
+   * @return true if vote was successfully added, otherwise false
    */
-  void addVerifiedVote(std::shared_ptr<Vote> const& vote);
+  bool addVerifiedVote(std::shared_ptr<Vote> const& vote);
 
   /**
    * @brief Check if the vote has been in the verified votes map
    * @param vote vote
    * @return true if exist
    */
-  bool voteInVerifiedMap(std::shared_ptr<Vote> const& vote);
+  bool voteInVerifiedMap(std::shared_ptr<Vote> const& vote) const;
+
+  /**
+   * @brief Inserts unique vote
+   * @param vote
+   * @return true if vote was successfully inserted(it was unique) or this specific vote was already inserted, otherwise
+   * false
+   */
+  bool insertUniqueVote(const std::shared_ptr<Vote>& vote);
```

Comment:
Why `insertUniqueVote` is separate function from `addVerifiedVote` ? is there a case where would you call just one? Because what I can see is that someone will call just one of those and break whole functionality

Reply:
the reason is that `insertUniqueVote` is called also when adding reward votes and next votes in NextVotesManager. The other solution is to separate this unique votes functionality into separate class and call it inside `addVerifiedVote`, `addRewardVote` and `updateWithSyncedVotes`

Reply:
https://github.com/Taraxa-project/taraxa-node/issues/1935